orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy

```diff
@@ -64,4 +73,10 @@ class UpsertApplicationTask extends AbstractFront50Task {
   String getNotificationType() {
     return "upsertapplication"
   }
+
+  private boolean isValidName(Application application) {
+    def cloudProviders = application?.cloudProviders ? application.cloudProviders.split(',') : []
+    boolean dashesAreValid = application.name.contains('-') ? cloudProviders?.every { it == 'kubernetes' } : true
```

Comment:
Would love an alternative to adding cloud provider-specific conditions into common orca tasks, we've got a couple already but should avoid adding more if possible (alternative would be an interface).

Reply:
I can refactor this out into another utils class so it's not baked into the task, sound OK?

Reply:
Should be an interface IMO rather than a Utils class that is called directly.